devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java

```diff
@@ -130,33 +132,23 @@ public Integer removeExtension(RunModeOption runMode, Set<String> extensions) th
     }
 
     @Override
-    public Integer info(boolean perModule) throws Exception {
-        ArrayDeque<String> args = new ArrayDeque<>();
-        setMavenProperties(args, true);
-        args.add("quarkus:info");
-        if (perModule) {
-            args.add("-DperModule");
-        }
-        args.add("-ntp");
-        return run(prependExecutable(args));
+    public Integer projectInfo(boolean perModule) throws Exception {
+        final ProjectInfo invoker = new ProjectInfo(quarkusProject());
+        invoker.perModule(perModule);
+        invoker.appModel(MavenProjectBuildFile.resolveApplicationModel(projectRoot));
+        return invoker.execute().isSuccess() ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
     }
 
     @Override
-    public Integer update(boolean rectify, boolean recommendedState, boolean perModule) {
-        ArrayDeque<String> args = new ArrayDeque<>();
-        setMavenProperties(args, true);
-        args.add("quarkus:update");
-        if (rectify) {
-            args.add("-Drectify");
-        }
-        if (recommendedState) {
-            args.add("-DrecommendedState");
-        }
-        if (perModule) {
-            args.add("-DperModule");
-        }
-        args.add("-ntp");
-        return run(prependExecutable(args));
+    public Integer updateProject(String targetPlatformVersion, String targetPlatformStreamId, boolean perModule)
+            throws Exception {
+        final UpdateProject invoker = new UpdateProject(quarkusProject());
+        invoker.latestCatalog(quarkusProject().getExtensionsCatalog());
+        //TODO resolve targetPlatformVersion from targetPlatformStreamId if needed
```

Comment:
@aloubyansky here

Reply:
and if both `targetPlatformVersion` and `targetPlatformStreamId` then we set to the latest